app/scripts/lib/rpc-method-middleware/handlers/get-provider-state.ts

```diff
@@ -0,0 +1,76 @@
+import type {
+  JsonRpcEngineNextCallback,
+  JsonRpcEngineEndCallback,
+} from '@metamask/json-rpc-engine';
+import type {
+  JsonRpcRequest,
+  JsonRpcResponse,
+  JsonRpcParams,
+  Json,
+} from '@metamask/utils';
+import { MESSAGE_TYPE } from '../../../../../shared/constants/app';
+import {
+  HandlerWrapperType,
+  getProviderStateType,
+  ProviderStateHandlerResult,
+} from './handlers-helper';
+
+type getProviderStateImplementationType<
+  Params extends JsonRpcParams = JsonRpcParams,
+  Result extends Json = Json,
+> = {
+  implementation: (
+    _req: JsonRpcRequest<Params>,
```

Comment:
There was no underscore in `req` from original implementation. I don't think it is a big deal, but just curious that if this is intentional?

Reply:
Oops, that was not intentional. Thank you for catching that :) I have updated that to the original implementation.